src/main/java/uk/gov/hmcts/reform/civil/service/stitching/BundleRequestExecutor.java

```diff
@@ -55,10 +56,13 @@ public CaseData post(final BundleRequest payload, final String endpoint, String
                 CaseDetails.class
             );
             if (response1.getStatusCode().equals(HttpStatus.OK)) {
-                return caseDetailsConverter.toCaseData(requireNonNull(response1.getBody()));
+                return Optional.of(caseDetailsConverter.toCaseData(requireNonNull(response1.getBody())));
             } else {
                 log.warn("The call to the endpoint with URL {} returned a non positive outcome (HTTP-{}). This may "
                              + "cause problems down the line.", endpoint, response1.getStatusCode().value());
+                log.info("Stitching endpoint returned {} with reason {}",
+                         response1.getStatusCodeValue(),
+                         response1.getStatusCode().getReasonPhrase());
```

Comment:
if you add - case id that will be good.

Reply:
Where exactly do I get it from 🙈 - Nigel has already got his ticket to add it to all logs

Reply:
From Payload -  if not then fine. :)